client/state/data-layer/wpcom/me/settings/profile-links/new/index.js

```diff
@@ -42,35 +39,33 @@ export const addUserProfileLinks = ( { dispatch }, action ) =>
  * - duplicate links
  * - malformed links
  *
- * @param   {Function} dispatch Redux dispatcher
- * @param   {Object}   action   Redux action
- * @param   {Array}    data     Response from the endpoint
+ * @param   {Object} action Redux action
+ * @param   {Array}  data   Response from the endpoint
  * @returns {Object} Dispatched user profile links add action
  */
-export const handleAddSuccess = ( { dispatch }, action, data ) => {
-	dispatch( addUserProfileLinksSuccess( data.profile_links ) );
+export const handleAddSuccess = ( action, data ) => {
+	const actions = [ addUserProfileLinksSuccess( action.profileLinks ) ];
 
 	if ( data.duplicate ) {
-		return dispatch( addUserProfileLinksDuplicate( data.duplicate ) );
+		actions.push( addUserProfileLinksDuplicate( data.duplicate ) );
+	} else if ( data.malformed ) {
```

Comment:
Does this need to be an `else` ? This is a slight change from the way the code was before.

Reply:
Good question. Yeah, this is intentional: we no longer `return` immediately; rather we collect actions in an array so we can later return them at the end. This is because this handler will always dispatch 2 actions (one for successful request, and one other that receives either the new profile links, or the malformed/duplicate ones).

Reply:
Ah yeah, now I see the `return` on `return dispatch( addUserProfileLinksDuplicate( data.duplicate ) );`, so the new code is equivalent 👍